src/protocol/vm/state.rs

```diff
@@ -422,6 +427,16 @@ impl VMPoolState<PreCachedDB> {
         Ok(limits?.0)
     }
 
+    fn clear_all_cache(&mut self, tokens: Vec<ERC20Token>) -> Result<(), SimulationError> {
+        self.clone()
```

Comment:
Why clone? We're have a `&mut` we should be able to clear the storage from it no?